### ISSUE TITLE
Failed to lookup address information: Name does not resolve

### ISSUE THREAD

**Issue by spaolacci** [codota/TabNine#319]
- OS version: Linux / Debian 10 "Buster"
 - Editor: vim
 - Editor version: 8.1
 - Programming language: Rust
 - TabNine extension version: 2.9.2

**Logs:** 
[2021-01-11][19:21:15][TabNine][INFO] Port 38843 was provided. Emulating YCMD.
[2021-01-11][19:21:15][lunar_transport::controller][INFO] Creating lunar client...
[2021-01-11][19:21:15][frontend::ycmd_emulation][INFO] Listening at 127.0.0.1:38843
[2021-01-11][19:21:15][lunar_local::bridge][INFO] TabNine Local communication thread started.
[2021-01-11][19:21:15][lunar_local::bridge][INFO] Loading model...
[2021-01-11][19:21:15][lunar_transport::controller][INFO] TabNine Cloud is disabled, communication thread is sleeping until configuration change...
[2021-01-11][19:21:15][lunar_local::locker][INFO] Trying to lock: "/tmp/model.lock"
[2021-01-11][19:21:15][lunar_local::locker][INFO] Lock acquired, performing the inner action
[2021-01-11][19:21:15][lunar_local::downloader][INFO] Starting download (last_failure=None).
[2021-01-11][19:21:15][lunar_local::downloader][INFO] Tried to find cached metadata but got 'No such file or directory (os error 2)', so downloading it instead...
[2021-01-11][19:21:15][updater][INFO] Making API request to https://api.tabnine.com/model_metadata/v1
[2021-01-11][19:21:15][lunar_local::downloader][ERROR] https://api.tabnine.com/model_metadata/v1: error trying to connect: failed to lookup address information: Name does not resolve

**Issue Details:**
TabNine fail to connect to tabnine.com so can't register or anything else. I did expect issues with http proxy (both upper and lower case https_proxy are defined) running within a corporate environment but failing on name resolution is much less expected.

Impact is mostly that I can't register (nor download anything of course).

Being a Rust developer, please tell me if there's any lib/framework I could locally to help investigate.

**Comment by shaiahr** [comment]
Hi
This sounds like some firewall or corporate DNS that is blocking the request. 
Can you ask your IT to check that?

**Comment by spaolacci** [comment]
Hi, thanks for the feedback.

The thing is that https://api.tabnine.com/ is not filtered by the proxy. If it was I would also expect some sort of "contents blocked" (or any such retrieval error) but much less a name resolution issue.

Failing at using the proxy seems a plausible reason for not being able to resolve. This might just boil down to assessing whether proxy env (or any other conf) are taken into consideration but I can't find straightforward way to find that information out of the logs (I've tried usual http{s}_proxy upper and lower case without success).

Is there any info/doc about how proxy details are to be passed in?

Thanks a lot,
Seb

**Comment by amircodota** [comment]
@spaolacci Tabnine will use HTTP_PROXY and HTTPS_PROXY environment variables if present.
Remember that Tabnine is run by your editor, so these environment variables need to be defined in such a way that the editor you are using will see them.

But I think you are right that if you fail to resolve api.tabnine.com then it will not help.

What happens if you run

`curl https://api.tabnine.com`

Do you still get the "Failed to resolve" error?

**Comment by spaolacci** [comment]
Hi,

Just to confirm `curl` and others standard tools have no issue at reaching the url (incl. this very browser).

As a (sort of) more precise check, I've verified that both `vim` `(:!env)` and TabNine proceses (outer and inner variants through `/proc/$pid/environ`) were having the two `HTTP_PROXY` and `HTTPS_PROXY` environment variables defined. I didn't manage to find anything suspicious here.

`strace`-ing the process in command line did show 1) access to environment variables 2) attempts at reaching my DNS (after opening `/etc/hosts` and `/etc/resolv.conf` confirming proxy info was just not taken into consideration).

It happens that issue was in the way the `HTTPS_PROXY` variable was populated. Replacing `HTTPS_PROXY=myproxy:8080` by `HTTPS_PROXY=http://myproxy:8080` (mind leading `http://`) did the trick. This is not that usual but here it is.

Everything is fine now, I managed to upgrade to 3.34-9 and register :).

Thanks a lot for your help.

Seb
